src/Razor/test/Microsoft.VisualStudio.Editor.Razor.Test/TelemetryReporterTests.cs

```diff
@@ -322,4 +322,108 @@ public void TrackLspRequest()
                 Assert.Equal(correlationId, correlationProperty.Value);
             });
     }
+
+    [Theory]
+    [InlineData(typeof(RemoteInvocationException), 3)]
+    [InlineData(typeof(OperationCanceledException), 3)]
+    [InlineData(typeof(AggregateException), 15000)]
+    public void ReportFault_MixedTypeNestedExceptions_NoStackOverflowException(Type exceptionType, int depth)
```

Comment:
FYI, `ReportFault_MixedTypeNestedExceptions_NoStackOverflowException` would throw stackoverflow exception for nested types RemoteInvocationException and OperationCanceledException when depth is larger